code/datums/components/on_hit_effect.dm

```diff
@@ -1,20 +1,27 @@
+#define HEY_SET_THIS_ARG "self-explainatory"
 /**
  * ## On Hit Effect Component!
  *
  * Component for other elements/components to rely on for on-hit effects without duplicating the on-hit code.
  * See Lifesteal, or bane for examples.
- *
- * THIS COULD EASILY SUPPORT COMPONENT_DUPE_ALLOWED but the getcomponent makes it throw errors. if you can figure that out feel free to readd the dupe types
  */
 /datum/component/on_hit_effect
+	dupe_mode = COMPONENT_DUPE_ALLOWED
 	///callback used by other components to apply effects
 	var/datum/callback/on_hit_callback
 	///callback optionally used for more checks
 	var/datum/callback/extra_check_callback
 	///optionally should we also apply the effect if thrown at something?
 	var/thrown_effect
-
-/datum/component/on_hit_effect/Initialize(on_hit_callback, extra_check_callback, thrown_effect = FALSE)
+	///The datum that loaded this component
+	var/datum/associated_datum
+
+/datum/component/on_hit_effect/Initialize(on_hit_callback, extra_check_callback, thrown_effect = FALSE, datum/associated_datum = HEY_SET_THIS_ARG)
+	if(associated_datum == HEY_SET_THIS_ARG)
+		stack_trace("[type] added to [parent.type] without a set 'associated_datum' argument (null is allowed)")
```

Comment:
This won't work the way you think it will work, if you pass it as `null` it will default to `HEY_SET_THIS_ARG`

![image](https://github.com/user-attachments/assets/997e275a-4b86-42b1-a05f-ac1199f5758e)

Reply:
Well, that's fucking interesting. Thanks for telling. I'll probably go for signals instead.